packages/expo-dev-launcher/bundle/screens/HomeScreen.tsx

```diff
@@ -254,48 +258,164 @@ function DevSessionList({ devSessions = [], onDevSessionPress }: DevSessionListP
 
           return (
-            <View key={app.url}>
+            <LoadingContainer key={app.id} isLoading={isLoading}>
               <Button.ScaleOnPressContainer
                 onPress={() => onAppPress(app.url)}
                 roundedTop={isFirst ? 'large' : 'none'}
                 roundedBottom={isLast ? 'large' : 'none'}
+                py="small"
                 bg="default">
-                <Row align="center" padding="medium" bg="default">
-                  <StatusIndicator size="small" status="success" />
-                  <Spacer.Horizontal size="small" />
-                  <View flex="1">
-                    <Button.Text color="default" numberOfLines={1}>
-                      {label}
-                    </Button.Text>
-                  </View>
-                  <ChevronRightIcon />
-                </Row>
+                {renderRow(app)}
               </Button.ScaleOnPressContainer>
               {!isLast && <Divider />}
-            </View>
+            </LoadingContainer>
           );
         })}
       </View>
     </View>
   );
 }
+
+function LoadingContainer({ children, isLoading }) {
+  const style = useLoadingContainerStyle(isLoading);
+  return <Animated.View style={style}>{children}</Animated.View>;
+}
+
+function RecentLocalPackagerRow({ label, url }) {
+  return (
+    <>
+      <Row align="center" px="medium" bg="default">
+        <StatusIndicator size="small" status="success" />
+        <Spacer.Horizontal size="small" />
+        <View flex="1">
+          <Button.Text color="default" numberOfLines={1}>
+            {label}
+          </Button.Text>
+        </View>
+        <ChevronRightIcon />
+      </Row>
+
+      <Row px="medium" align="center" bg="default">
+        <Spacer.Vertical size="tiny" />
+        <Spacer.Horizontal size="large" />
+        <Row
+          style={{
+            flexWrap: 'wrap',
+            flexShrink: 1,
+          }}>
+          <Text size="small" color="secondary" numberOfLines={1}>
+            {url}
+          </Text>
+        </Row>
+      </Row>
+    </>
+  );
+}
+
+function RecentEASUpdateRow({ label, url, branchName, message }) {
+  const palette = useExpoPalette();
+  return (
+    <View>
+      <Row align="center" px="medium" bg="default">
+        <Row
+          shrink="1"
+          style={{
+            backgroundColor: palette.blue['100'],
+          }}
+          py="tiny"
+          px="1.5"
+          rounded="medium"
+          align="center">
+          <BranchIcon
+            style={{ maxHeight: 10, maxWidth: 12, resizeMode: 'contain' }}
+            resizeMethod="scale"
+          />
+          <Spacer.Horizontal size="tiny" />
+          <View shrink="1">
+            <Text size="small" numberOfLines={1}>{`Branch: ${branchName}`}</Text>
+          </View>
+        </Row>
+
+        <Spacer.Horizontal />
+
+        <ChevronRightIcon />
+      </Row>
+
+      <Spacer.Vertical size="small" bg="default" />
+
+      {Boolean(message) && (
```

Comment:
this is now done - I think it looks good 

I added some logic to reuse the EAS Update information if it already exists in the cache (e.g when a user presses an update link in this list) otherwise we'll show the app name and update url